### DISCUSSION TITLE
Custom or Generalized Logs

### DISCUSSION THREAD

**Discussion by hlanderdev** [pocketbase/pocketbase#1179] [Q&A] [answered]
When using Pocketbase as a framework, It would be nice to have non-request logs integrated with the admin UI. As of right now, all of the errors from my custom endpoints are logged to the log database using the `ActivityLogger` middleware, but for project cohesion it would be nice to be able to log errors from other parts of the project to the admin UI as well, for example errors related to cron jobs.

I imagine this could be achieved by adding a general errors table to the log database, but this might not be desirable because it would only really be useful when using Pocketbase as a framework. The other way I imagine this being possible is if developers define their own tables to the log db, but this might be tricky to dynamically show them in the admin UI.

**Comment by ganigeorgiev** [comment, marked as answer]
There are too many things that are not clear for me with a general logging support.

For example:
- What logs?
- How the logs data will be stored - timestamp with serialized json?
- How they will be visualized?
- How rotation will be managed?
- How logs will be submitted - via HTTP API, dedicated cli command or direct insertion in the database? _(the last option may not work well since you can have only a single writer at a time and will require the external services to implement their own fail/retry mechanism)_

_For now this remains out of the scope for v1 because already a lot of other tasks have piled in the roadmap but I may revisit it when researching https://github.com/pocketbase/pocketbase/issues/1167 in the future._

**Reply by hlanderdev** [reply]
No problem. Sorry, I failed to notice #898. This would solve the issue I was talking about. It's less about providing a logging system, and more about being able to integrate custom logs to the admin UI. Either way, it seems this is a post v1 feature.
